src/models/room.ts

```diff
@@ -2199,8 +2199,9 @@ export class Room extends EventEmitter {
      *                   message events into the room.
      */
     public maySendMessage(): boolean {
-        return this.getMyMembership() === 'join' &&
-            this.currentState.maySendEvent(EventType.RoomMessage, this.myUserId);
+        return this.getMyMembership() === 'join' && this.client.isRoomEncrypted(this.roomId)
```

Comment:
should also check if the client has encryption enabled, otherwise the messages sent won't be encrypted and this check would be wrong